packages/nx-plugin/src/utils/versions.ts

```diff
@@ -1,20 +1,26 @@
-import { readFileSync } from 'node:fs';
-import { join } from 'node:path';
+import { readFile } from 'node:fs/promises';
+import { dirname, join } from 'node:path';
+import { fileURLToPath } from 'node:url';
 
-const workspaceRoot = join(__dirname, '../../');
-const projectsFolder = join(__dirname, '../../../');
-
-export const cpuNxPluginVersion = loadPackageJson(workspaceRoot).version;
-export const cpuModelVersion = loadPackageJson(
-  join(projectsFolder, 'cli'),
+const workspaceRoot = join(fileURLToPath(dirname(import.meta.url)), '../../');
+const projectsFolder = join(
+  fileURLToPath(dirname(import.meta.url)),
+  '../../../',
+);
+type PkgJson = { version: string };
+export const cpuNxPluginVersion = (await loadPackageJson(workspaceRoot))
+  .version;
+export const cpuModelVersion = (
+  await loadPackageJson(join(projectsFolder, 'cli'))
 ).version;
-export const cpuUtilsVersion = loadPackageJson(
-  join(projectsFolder, 'utils'),
+export const cpuUtilsVersion = (
+  await loadPackageJson(join(projectsFolder, 'utils'))
 ).version;
-export const cpuCliVersion = loadPackageJson(
-  join(projectsFolder, 'models'),
+export const cpuCliVersion = (
+  await loadPackageJson(join(projectsFolder, 'models'))
 ).version;
 
-function loadPackageJson(folderPath: string) {
-  return JSON.parse(readFileSync(join(folderPath, 'package.json')).toString());
+async function loadPackageJson(folderPath: string): Promise<PkgJson> {
+  const packageJsonContent = await readFile(join(folderPath, 'package.json'));
+  return JSON.parse(packageJsonContent.toString()) as PkgJson;
```

Comment:
If encoding is defined while reading a file, the method returns a string instead of a buffer. Source [here](https://stackoverflow.com/questions/6456864/why-does-node-jss-fs-readfile-return-a-buffer-instead-of-string).

```suggestion
async function loadPackageJson(folderPath: string): Promise<PkgJson> {
  const packageJsonContent = await readFile(join(folderPath, 'package.json'), 'utf8');
  return JSON.parse(packageJsonContent) as PkgJson;
```

Reply:
We have a `readJsonFile` in `utils` you could use instead.

Reply:
Good point, forgot about that one.

Reply:
used the helper